app/scripts/lib/createRPCMethodTrackingMiddleware.js

```diff
@@ -372,22 +427,23 @@ export default function createRPCMethodTrackingMiddleware({
         ...eventProperties,
         ...blockaidMetricProps,
         location,
-        // if security_alert_response from blockaidMetricProps is Benign, force set security_alert_reason to empty string
-        security_alert_reason:
-          blockaidMetricProps.security_alert_response ===
-          BlockaidResultType.Benign
-            ? ''
-            : blockaidMetricProps.security_alert_reason,
       };
 
-      trackEvent({
-        event,
-        category: MetaMetricsEventCategory.InpageProvider,
-        referrer: {
-          url: origin,
-        },
-        properties,
-      });
+      if (signatureUniqueId) {
+        finalizeSignatureFragment(metaMetricsController, req, {
+          abandoned: event === eventType.REJECTED,
+          properties,
+        });
```

Comment:
Hey @digiwand is event of type REJECTED received in this middleware ?

Reply:
hi @jpuri, yep! Within this next() node.js callback, the event is set based on the response above. This may include eventType.REJECTED